contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1605 @@
+            let { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "check-pox-addr-version",
+              [Cl.buffer(Uint8Array.from([version]))],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.BoolFalse));
+            expect(actual).toBeBool(expected);
+          }
+        )
+      );
+    });
+
+    it("should return true check-pox-lock-period for valid reward cycles number", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({ min: 1, max: 12 }),
+          (caller, reward_cycles) => {
+            // Arrange
+            const expected = true;
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "check-pox-lock-period",
+              [Cl.uint(reward_cycles)],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.BoolTrue));
+            expect(actual).toBeBool(expected);
+          }
+        )
+      );
+    });
+
+    it("should return false check-pox-lock-period for reward cycles number > 12", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({ min: 13 }),
+          (caller, reward_cycles) => {
+            // Arrange
+            const expected = false;
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "check-pox-lock-period",
+              [Cl.uint(reward_cycles)],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.BoolFalse));
+            expect(actual).toBeBool(expected);
+          }
+        )
+      );
+    });
+
+    it("should return false check-pox-lock-period for reward cycles number == 0", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          (caller) => {
+            // Arrange
+            const reward_cycles = 0;
+            const expected = false;
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "check-pox-lock-period",
+              [Cl.uint(reward_cycles)],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.BoolFalse));
+            expect(actual).toBeBool(expected);
+          }
+        )
+      );
+    });
+
+    it("should return (ok true) can-stack-stx for versions 0-4 valid pox addresses, hashbytes, amount, cycles number", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({ min: 0, max: 4 }),
+          fc.array(fc.nat({ max: 255 }), {
+            minLength: 20,
+            maxLength: 20,
```

Comment:
Are the array bounds really needed?

Reply:
Yes, they are needed because otherwise the function will return err. 
This happens because of this assertion from the pox-4 SC:
```
(asserts! (check-pox-addr-version (get version pox-addr))(err ERR_STACKING_INVALID_POX_ADDRESS))
```

This is not inside `can-stack-stx`, but in `minimal-can-stack-stx` which is called by the tested method.